mailbox/api/src/main/java/org/apache/james/mailbox/FlagsBuilder.java

```diff
@@ -34,21 +34,23 @@ public FlagsBuilder() {
     }
```

Comment:
ApplicableFlags indexing is a concern specific to Cassandra implementation.

The highest level this concern reaches is the indexTableUpdater. 

This changeset never reached the **mailbox/store** layer by the way. (proof that it is implementation specific concerns)

However, diff in the userFlags can be computed and passed by **CassandraIndexTableHandler.java**. Is that what you propose?